bindings/python/setup.py

```diff
@@ -187,24 +212,39 @@ def build_extension_with_b2(self):
         if self.cxxstd:
             args.append(f"cxxstd={self.cxxstd}")
 
-        # Jamfile hack to copy the module to our target directory
+        # Jamfile hacks to ensure we select the python environment defined in
+        # our project-config.jam
+        args.append("libtorrent-python=on")
+        args.append(f"python={sysconfig.get_python_version()}")
+
+        # Our goal is to produce an artifact at this path. If we do this, the
+        # distutils build system will skip trying to build it.
         target = pathlib.Path(self.get_ext_fullpath("libtorrent"))
+        self.announce(f"target: {target}")
+
+        # b2 doesn't provide a way to signal the name or paths of its outputs.
+        # We try to convince python.jam to name its output file like our target
+        # and copy it to our target directory. See comments in
+        # write_b2_python_config for limitations on controlling the filename.
+
+        # Jamfile hack to copy the module to our target directory
         args.append(f"python-install-path={target.parent}")
-        args.append("libtorrent-python=on")
         args.append("install_module")
 
         # We use a "project-config.jam" to instantiate a python environment
         # to exactly match the running one.
         config = tempfile.NamedTemporaryFile(mode="w+", delete=False)
         try:
-            write_b2_python_config(target, config)
+            write_b2_python_config(config)
             config.seek(0)
             self.announce("project-config.jam contents:")
             self.announce(config.read())
```

Comment:
what does `announce()` do? I thought initially that it would print it to the console, but this is not printed. (that's why I added the print of the config)

Reply:
It's a debug logger; it prints either with `DISTUTILS_DEBUG=1` or `setup.py -v`, I'm not sure which.